leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,17 +38,526 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
+        // }
+
+        // var repFuncSegment = this.functionSegList[repIndex];
+        // // remove RepFuncSegment object from array
+        // this.functionSegList.splice(repIndex, 1);
+
+        // // add the FuncSegments back into the array
+        // var j = repIndex;
+        // for (var i = 0; i < repFuncSegment.functionSegList.length; i++) {
+        //     this.functionSegList.splice(j, 0, repFuncSegment.functionSegList[i]);
+        //     j++;
+        // }
+    },    
+    
+    /**
+     * Returns the epoch boundary where the repeat segment
+     * starts
+     *
+     * Precondition: This EvolvingFunction must contain a RepFuncSegment
+     *
+     * @returns {String}
+     *   ex: 'A'
+     */
+    // TODO - fix this function 
+    getStartRepeatEpoch: function() {
+        if (this.repStart != 0) {
+            var repStartIndex = this.repStart.charCodeAt() - 64; // this should be the index of the first repeating segment
+            return this.functionSegList[repStartIndex].get('startTP');
+        }
+        else return this.functionSegList[0].get('startTP');
+
+        // Original Function
+        // for (var i = 0; i < this.functionSegList.length; i++) {
+        //     if (this.functionSegList[i] instanceof RepFuncSegment) {
+        //         return this.functionSegList[i].functionSegList[0].funcStart;
+        //     }
+        // }
+    },
+
+    /**
+     * Returns the epoch boundary where the repeat segment
+     * ends
+     *
+     * Precondition: This EvolvingFunction must contain a RepFuncSegment
+     *
+     * @returns {String}
+     *   ex: 'C'
+     */
+    // TODO - fix this function
+     getEndRepeatEpoch: function() {
+        var repStopIndex = this.repStop.charCodeAt() - 64; // this should be the index of the last repeating segment
+        return this.functionSegList[repStopIndex].get('stopTP');
+        // Original Function
+        // for (var i = 0; i < this.functionSegList.length; i++) {
+        //     if (this.functionSegList[i] instanceof RepFuncSegment) {
+        //         var len = this.functionSegList[i].functionSegList.length;
+        //         return this.functionSegList[i].functionSegList[len - 1].funcStop;
+        //     }
+        // }
+    },
+
+    /**
+     * Deleted Functions in EvolvingFunctionBBM
+     * All of these functions were either never used or are now not needed b/c of refactor
+     * 
+     * getRepeatAbsTime: function() {},
+     * getRepeatRepNum: function() {},
+     * hasRepeat: function() {}, 
+     * setAbsoluteTime: function(time) {},
+     * setRepNum: function(num) {},
+     * getFuncSegmentIterable: function() {},
+     * getRepFuncSegmentIndex: function() {},  
+     * getLastStopValue: function() {},
+     * getMarkedVal: function(i) {},  
+     * findSegmentByStartTime: function(time) {}, 
+     */
+
+
 });
 
 var IntentionBBM = Backbone.Model.extend({
     initialize: function(options) { 
         this.nodeType = options.nodeType;  
```

Comment:
This will prevent probably a lot of future errors - see message in slack surf21 channel
```suggestion
        _.extend({}, this.defaults, options)
```